### ISSUE TITLE
MP4 files not getting indexed

### ISSUE THREAD

**Issue by claudiobizzotto** [gerbera/gerbera#3074]
I've added an autoscan item to a folder but only the subfolders containing Matroska (.mkv) files get indexed. Subfolders containing MP4 files are ignored (not indexed).

0. Start SystemD service with vanilla config file.
1. Add an autoscan item to a parent folder. 
2. Wait for item to finish indexing.
3. No MP4 files (or folders containing them) show up.
4. No error messages, despite DEBUG=true

I would expect the MP4 files (and the folders containing them) to display just like the Matroska ones.

Server information:
 - OS: Ubuntu 22.04.4 LTS (Jammy)
 - Gerbera Version: 2.1.0
 - Versions of libraries: probably the latest
 - Source: Gerbera's [Ubuntu repo](https://github.com/gerbera/gerbera/blob/master/doc/install.rst#ubuntumint).

Media client information:
 - Device: Fedora desktop, iPhone, iPad.
 - Client app: VLC
 - Browser: N/A

I'm using the `--debug` flag when running the daemon, but the log messages don't say anything about why those MP4s are being ignored.

**Comment by KarlStraussberger** [comment]
We discovered some problems with mp4 in taglib recently. But this needs further investigation. Maybe, we are looking at ffmpeg data for this.

**Comment by claudiobizzotto** [comment]
Running now with both `--debug` and `--debug-mode="taglib"` but still no warnings or errors in the log file. Is there another way I may help debugging this?

**Comment by whyman** [comment]
So I think you have found a cause of the problem there, for some reason you getting the generic `octet-steam` mimetype, instead of for example `audio/mp4`, this means Gerbera will not treat the file as audio

**Comment by whyman** [comment]
Would be interesting to know what `file --mime /path/to/yourfile.mp4` gives you, and check that mime support is enabled in `gerbera --compile-info`

**Comment by claudiobizzotto** [comment]
```
$ file --mime /home/xxx/yyy/zzz/www.mp4
/home/xxx/yyy/zzz/www.mp4: video/mp4; charset=binary
```

As for mime support, I can see that `WITH_MAGIC=ON`:

```
$ gerbera --compile-info

Gerbera UPnP Server 2.1.0
Copyright 2016-2024 Gerbera Contributors.
Licence GPLv2: GNU GPL version 2.
This is free software: you are free to change and redistribute it.

Compile info
-------------
WITH_NPUPNP=OFF
WITH_MAGIC=ON
WITH_MYSQL=ON
WITH_CURL=ON
WITH_INOTIFY=ON
WITH_JS=ON
WITH_TAGLIB=ON
WITH_AVCODEC=ON
WITH_FFMPEGTHUMBNAILER=ON
WITH_EXIF=ON
WITH_EXIV2=ON
WITH_MATROSKA=ON
WITH_WAVPACK=ON
WITH_SYSTEMD=ON
WITH_LASTFM=OFF
WITH_DEBUG=ON
WITH_DEBUG_OPTIONS=ON
WITH_TESTS=OFF

Git info:
-------------
Git Branch: HEAD
Git Commit: a9bab24e043b5612a0264dbd753f85e1254a5873
```

**Comment by whyman** [comment]
Is there a symlink involved somehow?

**Comment by claudiobizzotto** [comment]
Regular files only, no symlinks.

**Comment by whyman** [comment]
Did you run the test magic command from inside the jail?

**Comment by claudiobizzotto** [comment]
Same result.

```
claudio@xxx:~$ sudo -u gerbera /bin/bash
gerbera@xxx:~$ file --mime /home/xxx/yyy/zzz/www.mp4
/home/xxx/yyy/zzz/www.mp4: video/mp4; charset=binary
```

**Comment by claudiobizzotto** [comment]
I still have Gerbera 1.12.1 stand-alone (no SystemD scripts) installed on my desktop and, when I run it against some local MP4 files, I can watch them through Gerbera on VLC.

Any chance this issue could be a regression introduced during the v2 bump? (I honestly don't recall seeing this problem before.)

**Comment by KarlStraussberger** [comment]
Did you already check out `extension-mimetype` mappings. If you updated from 1.12 it is also recommended to run `gerbera --create-config` and merge your settings back.

**Comment by claudiobizzotto** [comment]
I ran a diff against the new config and the old one and saw chances on some of the XML nodes (`extended-runtime-options`, `script-folder`, `import-function`, etc.) That looked promising and gave me hope for a few moments. The problem persists, though :(

**Comment by KarlStraussberger** [comment]
Good to know that your config.xml is up to day now.

We have no similar reports yet, so we have to narrow the problem down.
The missing detection of the mime type points to your chroot jail. Can you access `/usr/share/misc/magic` or whereever it is on your system.

**Comment by KarlStraussberger** [comment]
This is getting really strange, because the trace message you posted was moved a year ago

https://github.com/gerbera/gerbera/blob/a9bab24e043b5612a0264dbd753f85e1254a5873/src/content/import_service.cc#L532

The indicates that the version you are running is not 2.1.
Maybe there are multiple versions. Try
`type -a gerbera`

**Comment by claudiobizzotto** [comment]
Spot on! The issue is the jail. To confirm, I started a regular Gerbera process (v2) on my server, pointed the configs to the correct home folder, and voila: I can see a bunch of MP4s now.

Back to the original issue, the version used inside the jail was indeed an older one (1.2.1)... :disappointed: I updated it by repeating the entire Jail Kit recipe (`jk_cp` etc.) onto the jail, but still no MP4s. I suspect I'm missing some binary or `.so` files inside my jail (I've only copied the ones for `magic`, SQLite3 and Gerbera v2). But this is an issue I need to solve by myself, since mine's not a standard installation.

Closing this as this is not an issue with Gerbera. Thank you for all the helpful tips!
